submissions/Harnytskyi/js-pre-oop/index.js

```diff
@@ -0,0 +1,82 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/Harnytskyi/a-tiny-JS-world
+   Web app: https://harnytskyi.github.io/a-tiny-JS-world 
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+class Inhabitant {
+    constructor(species, name, gender, legs, hands, saying) {
+        this.species = species;
+        this.name = name;
+        this.gender = gender;
+        this.legs = legs;
+        this.hands = hands;
+        this.saying = saying;
+    };
+    toString() {
+        const infoMessage = [
+            this.saying,
+            'I am a',
+            this.species + '.',
+            'My name is',
+            this.name + '.',
+            'My gender is',
+            this.gender + '.',
+            `I have`,
+            this.legs,
+            this.definitionExtremities()
+        ];
+        return infoMessage.join(' ');
+    }
+    definitionExtremities() {
+        if (this.species == 'human')
+            return 'legs and ' + this.hands + ' hands.'
+        else
+            return 'paws.'
+    }
```

Comment:
Base class should know nothing about its children classes implementation.
Otherwise OOP and SOLID principles are broken.